### PR TITLE
Remove support for random_attributes, ua_strings, screen_res #745

### LINKED ISSUE #745: Remove support for random_attributes and browser_settings
These are a vestige of an unmaintained attempt to allow users to randomize some of the browser's properties. It's incomplete (#17), unsupported (#620), and untested.

Here are some code pointers:

https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L14-L35

https://github.com/mozilla/OpenWPM/blob/master/automation/DeployBrowsers/user_agent_strings.txt

https://github.com/mozilla/OpenWPM/blob/master/automation/DeployBrowsers/screen_resolutions.txt

https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/DeployBrowsers/deploy_firefox.py#L80-L97

https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/DeployBrowsers/deploy_firefox.py#L105-L110

Any future anti-bot circumvention work should be tracked in #448

### PR DESCRIPTION
**Pull request by ankushduacodes**
Running tests to fix #745

### CODE REVIEW

File: automation/BrowserManager.py [hunk 1]
```diff
@@ -111,7 +111,6 @@ def launch_browser_manager(self):
             # make sure browser loads crashed profile
             self.browser_params['recovery_tar'] = tempdir
             # don't re-randomize attributes
```
Comment (vringar) [inline automation/BrowserManager.py:113]:
```suggestion
            
```
Please also delete the associated comment

File: automation/default_browser_params.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "save_content": false,
     "callstack_instrument": false,
     "dns_instrument": false,
+    "bot_mitigation": false,
```
Comment (vringar) [inline automation/default_browser_params.json:11]:
```suggestion
    
```
The key already exists in line 18. Consider either moving it up here or removing this entry.

Comment (vringar) [reply automation/default_browser_params.json:11]: Also please remove the "random_attributes" in like 22.

**Review by vringar — CHANGES_REQUESTED** [review]
This PR is looking very good. But there are some little changes to be made.

**Comment by vringar** [comment]
You should also remove 
https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L13-L35
And the places where this code is called further down in the file.

File: automation/default_browser_params.json (hunk 2 above)
Comment (ankushduacodes) [reply automation/default_browser_params.json]: Could please explain what do you mean by `remove the "random_attributes" in like 22.`

**Comment by ankushduacodes** [comment]
> You should also remove
> https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L13-L35
> 
> 
> And the places where this code is called further down in the file.

Do you mean remove the comments? or the whole code snippet and its calls down the line?
if I need to remove the whole code snippet then, do I need to remove only these two functions and its calls? or is there other functions that I need to remove?

**Comment by vringar** [comment]
I mean the whole snippet and it's calls down the line as well as any references to browser_settings. browser_settings were only used to store ua_strings and screen_res and since we want to remove both, we don't need them anymore.

File: automation/BrowserManager.py [hunk 3]
```diff
@@ -75,7 +75,6 @@ def __init__(self, manager_params, browser_params) -> None:
         self.restart_required = False
 
         self.current_timeout: Optional[int] = None  # timeout of the current command
-        # dict of additional browser profile settings (e.g. screen_res)
         self.browser_settings = None
```
Comment (vringar) [inline automation/BrowserManager.py:78]: Meaning we can remove this line as well.

**Comment by ankushduacodes** [comment]
> I mean the whole snippet and it's calls down the line as well as any references to browser_settings. browser_settings were only used to store ua_strings and screen_res and since we want to remove both, we don't need them anymore.
> I don't think you need to remove any other functions but I might have missed something.

What do you suggest I do to remove browser_settings as in this case https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L143-L190
browser_settings are returned here https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/DeployBrowsers/deploy_firefox.py#L66-L82

**Comment by vringar** [comment]
> What do you suggest I do to remove browser_settings as in this case
> https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L143-L190
> 
Here I'd suggest deleting Lines 182 and 190
> 
> https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/DeployBrowsers/deploy_firefox.py#L66-L71
> 
> And here
> 
> https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/DeployBrowsers/deploy_firefox.py#L77-L82

Delete the assignment and keep the call. 
So change it to
```python
load_profile( 
     browser_profile_path, 
     manager_params, 
     browser_params, 
     browser_params["recovery_tar"], 
 ) 
```
And then go down in those functions to remove everything that mentions `profile_settings`

**Comment by ankushduacodes** [comment]
https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L59-L140
this code snippet  is unreachable due to the return statement above.
Do you want to also remove this code snippet or do you want to keep it?

**Comment by ankushduacodes** [comment]
@vringar Can you please briefly explain what exactly load_profile function is doing? and is 

As it is wrapped around with try-except block, we can only remove line 182 and line 190.

Is that try-except block needed after we delete the load_browser_settings function? at what point exactly are we expecting where an error may be needed to be handled?

To my understanding we are using load_profile function to load browser_settings and return the browser_settings and assigning it to profile_settings which we want to get rid of. and we are also extracting the tar.gz file into browser_profile_folder directory.

what exactly does full_profile.tar.gz or profile.tar.gz includes, I mean what kind of information?

**Comment by vringar** [comment]
So this is off the top of my head and might not be completely accurate, but as far as I remember:
> what exactly does full_profile.tar.gz or profile.tar.gz includes, I mean what kind of information?

These files contain a tarred Firefox Profile. This is where your cookies, history and settings are stored in one place.

We use Selenium to control Firefox. Selenium creates an empty profile folder in a temporary directory when we tell it to start a new Firefox instance.
But we want Firefox to use the profile we have had created somewhere else, so we ask it where that empty profile is and put the contents of the tar in there. This is what `load_profile` does.

Somehow we decided that we also needed to save some extra information for each browser which wasn't in the `browser_params` and not in the profile so we created `browser_settings` which contained things like screen size and just saved that right next to the tar.
But now we have changed our minds on this and have decided that the benefits of this aren't worth all of the complexity. So now we try to purge any usage of `browser_settings` to allow us to simplify `load_profile` even further in the future.

File: automation/DeployBrowsers/deploy_firefox.py [hunk 4]
```diff
@@ -237,6 +202,6 @@ def deploy_firefox(status_queue, browser_params, manager_params, crash_recovery)
     else:
         raise RuntimeError("Unable to identify Firefox process ID.")
 
-    status_queue.put(("STATUS", "Browser Launched", (int(pid), profile_settings)))
+    status_queue.put(("STATUS", "Browser Launched", (int(pid))))
```
Comment (vringar) [inline automation/DeployBrowsers/deploy_firefox.py:205]:
```suggestion
    status_queue.put(("STATUS", "Browser Launched", int(pid)))
```
As you only have one element here and no comma no tuple gets created.

File: automation/BrowserManager.py [hunk 5]
```diff
@@ -166,9 +162,7 @@ def check_queue(launch_status):
                 # 4. Browser launch attempted
                 check_queue(launch_status)
                 # 5. Browser launched
-                (self.geckodriver_pid, self.browser_settings) = check_queue(
-                    launch_status
-                )
+                (self.geckodriver_pid) = check_queue(launch_status)
```
Comment (vringar) [inline automation/BrowserManager.py:165]:
```suggestion
                self.geckodriver_pid = check_queue(launch_status)
```

**Review by vringar — APPROVED** [review]
Two small nitpicks but otherwise this looks good to me!

**Comment by vringar** [comment]
> https://github.com/mozilla/OpenWPM/blob/e989ce5f74c1737185c89a33bde4015b55df066a/automation/Commands/profile_commands.py#L59-L140
> 
> this code snippet is unreachable due to the return statement above.
> Do you want to also remove this code snippet or do you want to keep it?

Yes please keep this code around. We eventually want to re-enable profile dumping but haven't gotten around to it yet.

**Comment by ankushduacodes** [comment]
> So this is off the top of my head and might not be completely accurate, but as far as I remember:
> 
> > what exactly does full_profile.tar.gz or profile.tar.gz includes, I mean what kind of information?
> 
> These files contain a tarred Firefox Profile. This is where your cookies, history and settings are stored in one place.
> 
> We use Selenium to control Firefox. Selenium creates an empty profile folder in a temporary directory when we tell it to start a new Firefox instance.
> But we want Firefox to use the profile we have had created somewhere else, so we ask it where that empty profile is and put the contents of the tar in there. This is what `load_profile` does.
> 
> Somehow we decided that we also needed to save some extra information for each browser which wasn't in the `browser_params` and not in the profile so we created `browser_settings` which contained things like screen size and just saved that right next to the tar.
> But now we have changed our minds on this and have decided that the benefits of this aren't worth all of the complexity. So now we try to purge any usage of `browser_settings` to allow us to simplify `load_profile` even further in the future.
> 
> If any of this makes sense to you and you feel like you can shorten this a bit, please write this as a docstring on `load_profile`

Thank you for explaining, I think now have a big picture of what goes on when we save cookies in the browser.
Does this apply when a real user is using firefox, Is that how the cookies are stored in the system?

**Comment by vringar** [comment]
> Thank you for explaining, I think now have a big picture of what goes on when we save cookies in the browser.
> Does this apply when a real user is using firefox, Is that how the cookies are stored in the system?

Cookies are always stored in your profile. However usually your Profile just stays in one place on the system and doesn't get tarred or copied somewhere else.

**Comment by vringar** [comment]
Do you feel like there is anything else you'd want to do as part of this change?
Otherwise I'd be willing to merge it rn.

**Comment by ankushduacodes** [comment]
> Do you feel like there is anything else you'd want to do as part of this change?
> Otherwise I'd be willing to merge it rn.

I think I have made all the necessary changes. I think its good to go!...

**Comment by vringar** [comment]
Then thank you very much for following through on this!

**Pull request merged**